### PR TITLE
[CORL-508] api downloads by user id

### PR DESCRIPTION
**Pull request by tessalt**
## What does this PR do?
- adds a mutation for admins to request archive URL for a user's comment history
- adds a GET request for archive accepting token as query param

## How do I test this PR?

```
mutation requestComments($input: RequestUserCommentsDownloadInput!) {
  requestUserCommentsDownload(input:$input) {
    clientMutationId
    archiveURL
  }
}
```

### CODE REVIEW

File: src/core/server/app/handlers/api/account/download.ts [hunk 1]
```diff
@@ -216,6 +218,46 @@ export const downloadHandler = ({
   };
 };
 
+export const adminDownloadHandler = ({
+  mongo,
+  signingConfig,
+}: AdminDownloadOptions): RequestHandler => {
+  return async (req: Request, res, next) => {
```
Comment (tessalt) [inline src/core/server/app/handlers/api/account/download.ts:225]: the only difference between this method and the user-facing one is that it doesn't have the rate-limiting, I'm not mad about the duplicated code, but I could understand if someone was!

File: src/core/server/app/router/api/account.ts [hunk 2]
```diff
@@ -34,6 +35,7 @@ export function createNewAccountRouter(
   router.put("/invite", jsonMiddleware, inviteHandler(app));
 
   router.get("/downloadcheck", downloadCheckHandler(app));
+  router.get("/download", adminDownloadHandler(app));
```
Comment (tessalt) [inline src/core/server/app/router/api/account.ts:38]: I know @wyattjoh had opinions on the `/downloadcheck` endpoint but I'm happy with `/download?token=` for the GET request, so in that context maybe `/downloadcheck` makes more sense?

File: src/core/server/app/handlers/api/account/download.ts (hunk 1 above)
Comment (nick-funk) [reply src/core/server/app/handlers/api/account/download.ts:225]: I think we follow rule of 3, so this is acceptable.
As well, this is a pretty short handler, easy to mentally grok and maintain. :)

File: src/core/server/app/router/api/account.ts (hunk 2 above)
Comment (nick-funk) [reply src/core/server/app/router/api/account.ts:38]: Alternatively, if we have the requesting user from our auth token, maybe we can check if the user asking to download is an admin.

If they are admin, ignore rate limiting, else, rate limit.

**Review by nick-funk — APPROVED** [review]

File: src/core/server/app/router/api/account.ts (hunk 2 above)
Comment (tessalt) [reply src/core/server/app/router/api/account.ts:38]: I was just thinking a POST request doesn't make as much sense for the admin endpoint

**Comment by kgardnr** [comment]
Can I merge this one @tessalt / @nick-funk ?

**Comment by tessalt** [comment]
@kgardnr should be good!

**Pull request merged**